icu4c/source/common/BUILD

```diff
@@ -0,0 +1,361 @@
+# © 2021 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+
+# This file defines Bazel targets for ICU4C header and source files that are
+# used across i18n algorithm implementations and related tools.
+# The configuration of dependencies among targets is strongly assisted by the
+# file in deptest that maintains such information, at
+# icu4c/source/test/depstest/dependencies.txt .
+
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
+
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+# When compiling code in the `common` dir, the constant
+# `U_COMMON_IMPLEMENTATION` needs to be defined. See 
+# https://unicode-org.github.io/icu/userguide/howtouseicu#c-with-your-own-build-system .
+
+# If linker errors occur, then this may be a sign that the dependencies were 
+# not specified correctly. Use dependencies.txt in depstest for assistance. See
+# https://stackoverflow.com/q/66111709/2077918 .
+
+cc_library(
+    name = "headers",
+    hdrs = glob([
+        "unicode/*.h", # public
+        "*.h",         # internal
+        ],
+        exclude = ["norm2_nfc_data.h"],
+    ),
+    # need to add includes in order to preserve existing source files'
+    # include directives that use traditional paths, not paths relative to
+    # Bazel workspace
+    # https://stackoverflow.com/a/65635893/2077918
+    includes = ["."],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+cc_library(
+    name = "platform",
+    srcs = [
+        "cmemory.cpp", 
+        "uobject.cpp",
+        "cstring.cpp",
+        "cwchar.cpp",
+        "uinvchar.cpp",
+        "charstr.cpp",
+        "unistr.cpp",
+        "appendable.cpp",
+        "stringpiece.cpp",
+        "ustrtrns.cpp",
+        "ustring.cpp",  
+        "ustrfmt.cpp",  
+        "utf_impl.cpp",
+        "putil.cpp",
+        "ucln_cmn.cpp",  
+        "udataswp.cpp",  
+        "umath.cpp",
+        "umutex.cpp",
+        "sharedobject.cpp",
+        "utrace.cpp",
+    ],
+    deps = [
+        ":headers",
+        # omit other deps b/c they are sys symbols
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+        "U_ATTRIBUTE_DEPRECATED=",
```

Comment:
Why do we need to define U_ATTRIBUTE_DEPRECATED as empty? umachine.h defines it for gcc and MSVC and otherwise leaves it empty anyway. And it is indirectly used in definitions across the libraries, so defining it only for "platform" is inconsistent.